### PR TITLE
nh_fwd cache refresh packets need entropy

### LINKED ISSUE #476: nh_fwd cache refresh packets need entropy
When testing a single flow over several snabb instances  via ECMP, the vMX will use hash method to decide over which egress interface to send the cache refresh packet it has received from any of the Snabb instances. If this is just a single flow from the same IP source/destination pair, it will always pick the same one, which can be different from the one it has received it. 
While not strictly a problem in a real life deployment, adding some variance to the hash packet in form of a random source UDP/TCP port will greatly increase the likelyhood for nh_fwd to learn the next hop mac address quickly.

### PR DESCRIPTION
**Pull request by dpino**
Fixes #476.

### CODE REVIEW

File: src/apps/lwaftr/nh_fwd.lua [hunk 1]
```diff
@@ -582,39 +584,74 @@ local function test_ipv6_flow ()
...
-local function test_ipv4_cache_trigger ()
-   local pkt = packet.from_string(lib.hexundump([[
+local function test_ipv4_cache_trigger (pkt)
+   local ether_dhost = "52:54:00:00:00:01"
+   local refresh_packet = ipv4_cache_trigger(pkt, ethernet:pton(ether_dhost))
+   local eth_hdr = ethernet:new_from_mem(refresh_packet.data, ethernet_header_size)
+   local ip_hdr = ipv4:new_from_mem(refresh_packet.data + ethernet_header_size, pkt.length - ethernet_header_size)
+   assert(ip_hdr:src_eq(n_cache_src_ipv4))
+   assert(ethernet:ntop(eth_hdr:dst()) == ether_dhost)
+   lwdebug.print_pkt(refresh_packet)
```
Comment (kbara) [inline src/apps/lwaftr/nh_fwd.lua]: Is it intentional that you left this print in the test?

File: src/apps/lwaftr/nh_fwd.lua [hunk 2]
```diff
@@ -114,6 +120,10 @@ local function send_ipv6_cache_trigger (r, pkt, mac)
    transmit(r, ipv6_cache_trigger(pkt, mac))
 end
 
+local function random_port ()
+   return math.random(65535)
```
Comment (kbara) [inline src/apps/lwaftr/nh_fwd.lua]: That's a really unusually wide range. Take a look at https://www.cymru.com/jtk/misc/ephemeralports.html

File: src/apps/lwaftr/nh_fwd.lua (hunk 1 above)
Comment (dpino) [reply src/apps/lwaftr/nh_fwd.lua]: The task is still WIP (notice the [WIP] tag in the subject). I will remove the line when it's ready to merge. Thanks for noticing it anyway.

File: src/apps/lwaftr/nh_fwd.lua (hunk 2 above)
Comment (dpino) [reply src/apps/lwaftr/nh_fwd.lua]: The issue says "random source UDP/TCP port", for that I understand a random number in the 0-65535 range. Also if I restrict a port range maybe I'm penalizing certain softwires in the binding-table, so I prefer not to make any assumption.

**Comment by kbara** [comment]
LGTM

**Comment by dpino** [comment]
I didn't merge it before because I wanted to double-check, that despite all tests are passing, this feature didn't cause regressions.

Snabbvmx nexthop selftest is passing, which means packets arrive fine to the VM (with the right checksum) and a packet with the nexthop address comes out from the VM. Traffic is also processed correctly by snabbvmx, as shown below:

```bash
$ sudo ./snabb snabbvmx lwaftr --id xe0 \
   --conf ./program/snabbvmx/tests/conf/snabbvmx-lwaftr-xe0.cfg \
   --pci 02:00.0 --mac 02:aa:aa:aa:aa:aa --sock /tmp/vh1a.sock --mirror tap0
```

```bash
$ sudo ./snabb packetblaster replay ./program/snabbvmx/tests/pcap/input/v4v6-256.pcap 82:00.0
```

```
$ sudo ./snabb snabbvmx top 3822

lwaftr (rx/tx/txdrop in Mpps)                       rx      tx      rxGb    txGb    txdrop     
lwaftr_v6                                           1.31    0.87    5.75    4.11    0.000000   
lwaftr_v4                                           0.87    1.31    3.83    5.33    0.000000   
```

**Pull request merged**